### PR TITLE
WP & PHP Upgrade Notices

### PR DESCRIPTION
**Pull request by kloon**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
This PR introduces a dashboard widget and a banner that notifies and prompts users to upgrade their WordPress and/or PHP if they are outdated and not part of the minimum required versions going into WooCommerce 3.7.

This PR needs to go out in the 3.6 release.

Dashboard Widget

Banner

### How to test the changes in this Pull Request:

1. Checkout branch
2. Update the WC_MIN_PHP_VERSION and/or WC_MIN_WP_VERSION constants in class-woocommerce.php to versions higher than your current WP and PHP versions.
3. Check the dashboard widget loads and the banner is displaying
4. Test dismissing the banner and ensuring the widget also does not load anymore.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> New - WordPress & PHP upgrade nudges when running older versions.

### CODE REVIEW

**Review by claudiosanches — APPROVED** [review]
Great job!

**Comment by kloon** [comment]
Thanks for review @claudiosanches, going to move this to In Progress again since the banner needs some changes. See p6q8Tx-XP-p2

**Review by rodrigoprimo — COMMENTED** [review]
I left some inline comments with some minor issues that we might want to address before merging this.

Another thing is that the notice is that if the user is running an unsupported WP version and the notice is displayed, it will keep displaying even after WP is updated (I didn't check with a PHP update but I assume the same will happen). Maybe we need to change the code to remove the notice after the update?

File: includes/admin/class-wc-admin-notices.php [hunk 1]
```diff
@@ -373,6 +380,26 @@ public static function wc_admin_feature_plugin_notice() {
...
 	}
 
+	/**
+	 * Notice about WordPress and PHP minimum requirements.
+	 *
+	 * @since 3.6.5
+	 * @return void
+	 */
+	public static function wp_php_min_requirements_notice() {
+		global $wp_version;
```
[the remark this answers is not in the document]
Comment (claudiosanches) [reply includes/admin/class-wc-admin-notices.php]: I prefer use `get_bloginfo()`, always nice avoid calling globals

**Comment by peterfabian** [comment]
As it's currently implemented, the refresh of displaying/not displaying the notice only happens on theme switch (`switch_theme`) and WooCommerce install (`woocommerce_installed`). 
We could perhaps hook into WP update, but afaik there is no way to hook into PHP version update in the context of WP, thus we need to check for WP and PHP versions also in the notice function itself (`wp_php_min_requirements_notice`), not only when adding the notice. This should cover cases when updating WP and/or PHP.

Theoretically speaking, the notice would not show if you had 'up to date' versions of WP/PHP, but then decided to downgrade them, but I would guess it's quite an edge case, so I decided to keep the add min notice a part of `reset_admin_notices`.

**Comment by peterfabian** [comment]
New message formats:
Screenshot 2019-06-26 at 16 41 29

Screenshot 2019-06-26 at 16 42 02

Screenshot 2019-06-26 at 16 44 01

File: includes/admin/class-wc-admin-notices.php [hunk 2]
```diff
@@ -373,6 +375,67 @@ public static function wc_admin_feature_plugin_notice() {
 		include dirname( __FILE__ ) . '/views/html-notice-wc-admin.php';
 	}
 
+	/**
+	 * Add notice about minimum PHP and WordPress requirement.
+	 *
+	 * @since 3.6.5
+	 */
+	public static function add_min_version_notice() {
+		$php_version = phpversion();
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-notices.php]: We don't really need to set those variables here, since we only use once.

**Review by rodrigoprimo — APPROVED** [review]
Thanks for the changes, Peter. Looks good to me now.

**Pull request merged**